src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,134 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+
 public class StringManipulation implements StringManipulationInterface {
+    String string = null;
+    // Default constructor
+    StringManipulation() {} 
+ 
+
+    // Parameterized constructor
+    StringManipulation(String string) {
+        this.string = string;
+    }
 
     @Override
     public String getString() {
-        return null;
+        return this.string;
     }
 
     @Override
     public void setString(String string) {
+        this.string = string;
     }
 
+    // todo: raise nullPointer Exception if string is null
     @Override
     public int count() {
-        return 0;
+        int count = 0;
+        String string = getString();
+        char ch[]= new char[string.length()];     
+        for(int i=0;i<string.length();i++)  
+        {  
+            ch[i]= string.charAt(i);  
+            if(((i > 0) && (ch[i] != ' ') && (ch[i-1] == ' ')) || ((ch[0] != ' ') && (i == 0)))  
+                count++;
+        }
+        return count;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        // throw IllegalArgument Exception if n <= 0
+        if (n <= 0) {
+            throw new IllegalArgumentException("n has to be at least 1");
+        }
+
+        String string = this.getString();
```

Comment:
I see in the last method, you called getString();.  I think you could do the same here and do not need this.